From e1979cfcb2a8d7d05a93f32a95cd34a48f999e89 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Mon, 24 Oct 2022 14:29:18 +0300
Subject: [PATCH] [SW-106067] tpm: use common defs

Use common TPM definitions provided by embedded-specs, instead of
duplicating them.

Change-Id: I86bec1e7b8ec646db4427b587a6277609d207759
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/arch/gaudi2/common/tpm2.c b/arch/gaudi2/common/tpm2.c
index 0517b03..9388c42 100644
--- a/arch/gaudi2/common/tpm2.c
+++ b/arch/gaudi2/common/tpm2.c
@@ -19,84 +19,6 @@
 #include "lzo/unaligned.h"
 #include "tpm_internal.h"
 
-/**
- * TPM2 Structure Tags for command/response buffers.
- *
- * @TPM2_ST_NO_SESSIONS: the command does not need an authentication.
- * @TPM2_ST_SESSIONS: the command needs an authentication.
- */
-enum tpm2_structures {
-	TPM2_ST_NO_SESSIONS	= 0x8001,
-	TPM2_ST_SESSIONS	= 0x8002,
-};
-
-/**
- * TPM2 command codes used at the beginning of a buffer, gives the command.
- *
- * @TPM2_CC_STARTUP: TPM2_Startup().
- * @TPM2_CC_SELF_TEST: TPM2_SelfTest().
- * @TPM2_CC_CLEAR: TPM2_Clear().
- * @TPM2_CC_CLEARCONTROL: TPM2_ClearControl().
- * @TPM2_CC_HIERCHANGEAUTH: TPM2_HierarchyChangeAuth().
- * @TPM2_CC_PCR_SETAUTHPOL: TPM2_PCR_SetAuthPolicy().
- * @TPM2_CC_DAM_RESET: TPM2_DictionaryAttackLockReset().
- * @TPM2_CC_DAM_PARAMETERS: TPM2_DictionaryAttackParameters().
- * @TPM2_CC_GET_CAPABILITY: TPM2_GetCapibility().
- * @TPM2_CC_GET_RANDOM: TPM2_GetRandom().
- * @TPM2_CC_PCR_READ: TPM2_PCR_Read().
- * @TPM2_CC_PCR_EXTEND: TPM2_PCR_Extend().
- * @TPM2_CC_PCR_SETAUTHVAL: TPM2_PCR_SetAuthValue().
- */
-enum tpm2_command_codes {
-	TPM2_CC_STARTUP		= 0x0144,
-	TPM2_CC_SELF_TEST	= 0x0143,
-	TPM2_CC_CLEAR		= 0x0126,
-	TPM2_CC_CLEARCONTROL	= 0x0127,
-	TPM2_CC_HIERCHANGEAUTH	= 0x0129,
-	TPM2_CC_PCR_ALLOCATE	= 0x012B,
-	TPM2_CC_PCR_SETAUTHPOL	= 0x012C,
-	TPM2_CC_DAM_RESET	= 0x0139,
-	TPM2_CC_DAM_PARAMETERS	= 0x013A,
-	TPM2_CC_NV_READ         = 0x014E,
-	TPM2_CC_GET_CAPABILITY	= 0x017A,
-	TPM2_CC_GET_RANDOM      = 0x017B,
-	TPM2_CC_PCR_READ	= 0x017E,
-	TPM2_CC_PCR_EXTEND	= 0x0182,
-	TPM2_CC_PCR_SETAUTHVAL	= 0x0183,
-};
-
-enum tpm_timeout {
-	TPM_TIMEOUT_MS			= 5,
-	TIS_SHORT_TIMEOUT_MS		= 750,
-	TIS_LONG_TIMEOUT_MS		= 2000,
-	SLEEP_DURATION_US		= 60,
-	SLEEP_DURATION_LONG_US		= 210,
-};
-
-enum {
-	TPM_STS_FAMILY_SHIFT		= 26,
-	TPM_STS_FAMILY_MASK		= 0x3 << TPM_STS_FAMILY_SHIFT,
-	TPM_STS_FAMILY_TPM2		= 1 << TPM_STS_FAMILY_SHIFT,
-	TPM_STS_RESE_TESTABLISMENT_BIT	= 1 << 25,
-	TPM_STS_COMMAND_CANCEL		= 1 << 24,
-	TPM_STS_BURST_COUNT_SHIFT	= 8,
-	TPM_STS_BURST_COUNT_MASK	= 0xffff << TPM_STS_BURST_COUNT_SHIFT,
-	TPM_STS_VALID			= 1 << 7,
-	TPM_STS_COMMAND_READY		= 1 << 6,
-	TPM_STS_GO			= 1 << 5,
-	TPM_STS_DATA_AVAIL		= 1 << 4,
-	TPM_STS_DATA_EXPECT		= 1 << 3,
-	TPM_STS_SELF_TEST_DONE		= 1 << 2,
-	TPM_STS_RESPONSE_RETRY		= 1 << 1,
-};
-
-enum tis_access {
-	TPM_ACCESS_VALID		= 0x80,
-	TPM_ACCESS_ACTIVE_LOCALITY	= 0x20,
-	TPM_ACCESS_REQUEST_PENDING	= 0x04,
-	TPM_ACCESS_REQUEST_USE		= 0x02,
-};
-
 /* Number of wait states to wait for */
 #define TPM_WAIT_STATES 100
 
@@ -144,14 +66,6 @@
  * @TPM2_RH_ENDORSEMENT: refers to the 'endorsement' hierarchy.
  * @TPM2_RH_PLATFORM: refers to the 'platform' hierarchy.
  */
-enum tpm2_handles {
-	TPM2_RH_OWNER		= 0x40000001,
-	TPM2_RS_PW		= 0x40000009,
-	TPM2_RH_LOCKOUT		= 0x4000000A,
-	TPM2_RH_ENDORSEMENT	= 0x4000000B,
-	TPM2_RH_PLATFORM	= 0x4000000C,
-};
-
 enum {
 	/* Secure storage for firmware settings */
 	TPM_HT_PCR = 0,
diff --git a/arch/gaudi2/common/tpm_internal.h b/arch/gaudi2/common/tpm_internal.h
index b03a86b..4d4af81 100644
--- a/arch/gaudi2/common/tpm_internal.h
+++ b/arch/gaudi2/common/tpm_internal.h
@@ -6,23 +6,6 @@
 #ifndef __tpm_internal_h
 #define __tpm_internal_h
 
-enum {
-	TPM_MAX_ORDINAL			= 243,
-	TPM_MAX_PROTECTED_ORDINAL	= 12,
-	TPM_PROTECTED_ORDINAL_MASK	= 0xff,
-	TPM_CMD_COUNT_BYTE		= 2,
-	TPM_CMD_ORDINAL_BYTE		= 6,
-};
-
-enum tpm_duration {
-	TPM_SHORT = 0,
-	TPM_MEDIUM = 1,
-	TPM_LONG = 2,
-	TPM_UNDEFINED,
-
-	TPM_DURATION_COUNT,
-};
-
 static unsigned int duration_ms[TPM_DURATION_COUNT];
 
 /*
diff --git a/include/tpm2.h b/include/tpm2.h
index 74c12e1..0932f7c 100644
--- a/include/tpm2.h
+++ b/include/tpm2.h
@@ -10,59 +10,9 @@
 
 #include <types.h>
 #include <stdbool.h>
+#include <hl_tpm2.h>
 
 #define TPM2_SHA384_DIGEST_SIZE 48
-
-/**
- * TPM2 startup values.
- *
- * @TPM2_SU_CLEAR: reset the internal state.
- * @TPM2_SU_STATE: restore saved state (if any).
- */
-enum tpm2_startup_types {
-	TPM2_SU_CLEAR		= 0x0000,
-	TPM2_SU_STATE		= 0x0001,
-};
-
-/**
- * TPM2 return codes.
- */
-enum tpm2_return_codes {
-	TPM2_RC_SUCCESS		= 0x0000,
-	TPM2_RC_BAD_TAG		= 0x001E,
-	TPM2_RC_FMT1		= 0x0080,
-	TPM2_RC_HASH		= TPM2_RC_FMT1 + 0x0003,
-	TPM2_RC_VALUE		= TPM2_RC_FMT1 + 0x0004,
-	TPM2_RC_SIZE		= TPM2_RC_FMT1 + 0x0015,
-	TPM2_RC_BAD_AUTH	= TPM2_RC_FMT1 + 0x0022,
-	TPM2_RC_HANDLE		= TPM2_RC_FMT1 + 0x000B,
-	TPM2_RC_VER1		= 0x0100,
-	TPM2_RC_INITIALIZE	= TPM2_RC_VER1 + 0x0000,
-	TPM2_RC_FAILURE		= TPM2_RC_VER1 + 0x0001,
-	TPM2_RC_DISABLED	= TPM2_RC_VER1 + 0x0020,
-	TPM2_RC_AUTH_MISSING	= TPM2_RC_VER1 + 0x0025,
-	TPM2_RC_COMMAND_CODE	= TPM2_RC_VER1 + 0x0043,
-	TPM2_RC_AUTHSIZE	= TPM2_RC_VER1 + 0x0044,
-	TPM2_RC_AUTH_CONTEXT	= TPM2_RC_VER1 + 0x0045,
-	TPM2_RC_NEEDS_TEST	= TPM2_RC_VER1 + 0x0053,
-	TPM2_RC_WARN		= 0x0900,
-	TPM2_RC_TESTING		= TPM2_RC_WARN + 0x000A,
-	TPM2_RC_REFERENCE_H0	= TPM2_RC_WARN + 0x0010,
-	TPM2_RC_LOCKOUT		= TPM2_RC_WARN + 0x0021,
-};
-
-/**
- * TPM2 algorithms.
- */
-enum tpm2_algorithms {
-	TPM2_ALG_SHA1           = 0x04,
-	TPM2_ALG_XOR            = 0x0A,
-	TPM2_ALG_SHA256         = 0x0B,
-	TPM2_ALG_SHA384         = 0x0C,
-	TPM2_ALG_SHA512         = 0x0D,
-	TPM2_ALG_NULL           = 0x10,
-	TPM2_ALG_SM3_256        = 0x12,
-};
 
 int tpm2_init(bool print);
 int tpm2_startup(enum tpm2_startup_types mode);
